src/libos/src/fs/disk_file.rs

```diff
@@ -131,6 +134,25 @@ impl DiskFile {
     pub fn ioctl(&self, _cmd: &mut dyn IoctlCmd) -> Result<()> {
         return_errno!(EINVAL, "this file does not support ioctl");
     }
+
+    pub fn metadata(&self) -> Metadata {
+        Metadata {
+            dev: 0,
+            inode: 0x1000,
+            size: self.disk.total_bytes(),
+            blk_size: BLOCK_SIZE,
+            blocks: self.disk.total_blocks(),
```

Comment:
According to the man page, the block number counts 512B blocks (i.e., sectors), not FS blocks.

```
blkcnt_t  st_blocks;  /* number of 512B blocks allocated */
```

Reply:
Good suggestion, I presume that the bocks should be the number of block_size, but it should be the number of 512B